### PR TITLE
Performance improvements, mostly to EditsMulti

### PR DESCRIPTION
**Pull request by benjaminjkraft**
It was doing a bunch of unnecessary work; now it's not!  There is more
that could be done to avoid generating the same edit multiple times, but
that would break compatibility since these are public methods, so I did
not do so at this time.

The added benchmark is now faster by a bit over 2x, and allocations
are reduced similarly.
```
before             92659             12744 ns/op           12736 B/op        139 allocs/op
after             214692              6158 ns/op            6816 B/op         66 allocs/op
```

I made a few other improvements where I noticed them.

### CODE REVIEW

File: fuzzy.go [hunk 1]
```diff
@@ -355,34 +358,21 @@ func (model *Model) EditsMulti(term string, depth int) []string {
 			break
 		}
 		for _, edit := range edits {
-			edits2 := Edits1(edit)
```
Comment (benjaminjkraft) [inline fuzzy.go:358]: to my surprise, this doesn't actually have any perf benefit, but I found it cleaner so I kept it.

**Comment by benjaminjkraft** [comment]
I believe the test failure is the bug fixed in #23; tests pass locally.

**Comment by benjaminjkraft** [comment]
Also, for reference, I have a more efficient, but more complex [implementation](https://github.com/sajari/fuzzy/compare/master...Khan:benkraft.perf2?expand=1), which reorders and deduplicates the list.  It's another 3x faster and another 50% fewer allocations.  Happy to submit that if desired.
```
after             628777              2122 ns/op            1616 B/op         39 allocs/op
```

**Comment by mish15** [comment]
@benjaminjkraft thanks for this. Looks like it's failing CI (unrelated, which will be fixed in #23 once complete). Will get this looked at once that is merged.

**Comment by benjaminjkraft** [comment]
Ok, with that plus a fix for a flaky test, it looks like it's passing now!

**Comment by mish15** [comment]
LGTM

**Pull request merged**